src/core/temporal/qgsabstracttemporal.cpp

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+                         qgsabstracttemporal.cpp
+                         ---------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsabstracttemporal.h"
+
+QgsAbstractTemporal::QgsAbstractTemporal()
+{
+}
+
+QgsAbstractTemporal::QgsAbstractTemporal( const bool enabled )
+  : mTemporal( enabled )
+{
+}
+
+void QgsAbstractTemporal::setIsTemporal( bool enabled )
+{
+  mTemporal = enabled;
+}
+
+bool QgsAbstractTemporal::isTemporal() const
+{
+  return mTemporal;
+}
+
+void QgsAbstractTemporal::setTemporalRange( const QgsDateTimeRange &dateTimeRange )
+{
+  if ( !isTemporal() )
```

Comment:
I don't think we should have this precondition -- we should still allow the temporal range to be set/retrieved even if the temporal attributes are disabled.

Reply:
Ok

Reply:
By the way @nyalldawson, removing the precondition will allow setting render context temporal range even if it not a temporal one!!